loadbalancer.go

```diff
@@ -31,10 +31,9 @@ func (lb RandomLoadBalancer[T]) Name() LoadBalancerPolicy {
 
 // Resolve return the resolved option for Random LB
 func (lb RandomLoadBalancer[T]) Resolve(dbs []T) T {
-	if len(lb.randInt) == 0 {
+	for len(lb.randInt) == 0 {
```

Comment:
why though?

Reply:
@bxcodec its necessary to have a `predict` call for every `Resolve` call

Reply:
To prevent deadlocks during, mulitthreading. 

I presume what's the happening here #45 , `predicts` call are not being made due to concurrency.

Reply:
@bxcodec updated the pr about.